### PR TITLE
drivers: pinctrl: Microchip XEC PINCTRL add invert pin

### PR DESCRIPTION
**Pull request by jvasanth1**
Add support for inverting the output (of alternate pin functions) on Microchip XEC GPIO pins.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

### CODE REVIEW

File: dts/bindings/pinctrl/microchip,xec-pinctrl.yaml [hunk 1]
```diff
@@ -119,3 +119,10 @@ child-binding:
             is "1x" for most pins. PIO pins are 2, 4, 8, or 12 mA. PIO-24 pins
             are 4, 8, 16, or 24 mA. Please refer to the data sheet for each
             pin's PIO type and default drive strength.
+
+        output-func-invert:
```
Comment (gmarull) [inline dts/bindings/pinctrl/microchip,xec-pinctrl.yaml:123]: please prefix vendor specific props, `microchip,output-func-invert`.

Comment (jvasanth1) [reply dts/bindings/pinctrl/microchip,xec-pinctrl.yaml]: updated accordingly

**Review by gmarull — APPROVED** [review]

**Comment by jvasanth1** [comment]
@albertofloyd can you help to review the PR?

**Comment by jvasanth1** [comment]
@MaureenHelm can you help to review and approve this PR?

**Review by albertofloyd — APPROVED** [review]

**Pull request merged**